noir_stdlib/src/ec/tecurve.nr

```diff
@@ -227,34 +227,16 @@ mod curvegroup {
 
         // Check for equality
         fn eq(self, p: Point) -> bool {
-            if self.is_zero() == true {
-                p.is_zero()
-            } else if p.is_zero() == true {
-                false
-            } else {
-                let Self {x: x1, y: y1, t: _t1, z: z1} = self;
-                let Self {x: x2, y: y2, t: _t2, z:z2} = p;
-                
-                if x1*z2 == x2*z1 {
-                    y1*z2 == y2*z1
-                } else {
-                    false
-                }
-            }
+            let Self {x: x1, y: y1, t: _t1, z: z1} = self;
+            let Self {x: x2, y: y2, t: _t2, z:z2} = p;
+
+            (x1*z2 == x2*z1) & (y1*z2 == y2*z1)
```

Comment:
Where does `t1*z2 == t2*z1` come from?

Reply:
This comes from the fact that we are dealing with projective coordinates. On second thought, this is only really valid if `z != 0`, which would be the case for points on the curve, and if the points are constructed properly, we would also automatically have `t1*z2 == t2*z1`, so this would fall under the same disclaimer as the zero check. This check is not present in the corresponding Arkworks implementation for example.